### PR TITLE
CakePHP 5 changes

### PR DESCRIPTION
**Pull request by LordSimal**
This should go into a separate `cake5` or `4.x` branch.

### CODE REVIEW

File: src/Event/FootprintListener.php [hunk 1]
```diff
@@ -3,10 +3,12 @@
 
 namespace Muffin\Footprint\Event;
 
+use ArrayObject;
 use Cake\Core\InstanceConfigTrait;
 use Cake\Datasource\EntityInterface;
 use Cake\Event\EventInterface;
 use Cake\Event\EventListenerInterface;
+use Cake\ORM\Query;
```
Comment (ADmad) [inline src/Event/FootprintListener.php:11]: This needs to be `SelectQuery`.

**Comment by ADmad** [comment]
One final change, can you please rename `Plugin` class to `FootprintPlugin`.

**Comment by ADmad** [comment]
Why did the rename start throwing stan errors, it was green before 😆

**Comment by ADmad** [comment]
Muchas gracias.

**Comment by LordSimal** [comment]
De nada.

**Pull request merged**